### PR TITLE
Preparation for release v2024.1 (Zonemaster/Zonemaster)

### PR DESCRIPTION
**Pull request by matsduf**
## Purpose

Preparation for release.

## Changes

Updates file `Changes` and other files. Updates to Zonemaster-GUI installation instructions.

## How to test this PR

Review the changes.

### CODE REVIEW

**Review by marc-vanderwal — DISMISSED** [review]

File: docs/public/RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+### [Zonemaster product]
+This version of Zonemaster also consists of the following components. For each component, see its Changes file or Github release notes. This fix version does not change anything in other components. See previous releases for recent changes.
+
+Component | Github release notes | Changes file | Updated in this release
+------------------------|:-----------------------:|---------------------------------|:----:
+[Zonemaster-LDNS]      | [v4.0.2][ldns-tag]        | [Changes][ldns-Changes]          | Yes
+[Zonemaster-Engine]    | [v6.0.0][engine-tag]    | [Changes][engine-Changes]       | No
+[Zonemaster-CLI]          | [v7.0.0][cli-tag]           | [Changes][cli-Changes]              | No
+[Zonemaster-Backend] | [v11.2.0][backend-tag] | [Changes][backend-Changes]   | Yes
+[Zonemaster-GUI]         | [v4.3.0][gui-tag]          | [Changes][gui-Changes]            | No
```
Comment (mattias-p) [inline docs/public/RELEASE.md:24]: How about making the table pretty in text form too?

```suggestion
Component            | Github release notes   | Changes file               | Updated in this release
---------------------|:----------------------:|----------------------------|:----------------------:
[Zonemaster-LDNS]    | [v4.0.2][ldns-tag]     | [Changes][ldns-Changes]    | Yes
[Zonemaster-Engine]  | [v6.0.0][engine-tag]   | [Changes][engine-Changes]  | No
[Zonemaster-CLI]     | [v7.0.0][cli-tag]      | [Changes][cli-Changes]     | No
[Zonemaster-Backend] | [v11.2.0][backend-tag] | [Changes][backend-Changes] | Yes
[Zonemaster-GUI]     | [v4.3.0][gui-tag]      | [Changes][gui-Changes]     | No
```

File: docs/public/RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
+# Release v2024.1 (2024-07-01)
+
+### [Release information]
+ - Translations have not been fully updated in this release. They will be updated in an upcoming release.
+
+### [Breaking changes]
```
Comment (mattias-p) [inline docs/public/RELEASE.md:6]: https://github.com/zonemaster/zonemaster-engine/pull/1339 is also breaking.

Comment (matsduf) [reply docs/public/RELEASE.md:6]: In what way is it breaking?

Comment (mattias-p) [reply docs/public/RELEASE.md:6]: The profile.json file in the dist directory used to be part of the public interface of Zonemaster Engine but it isn't anymore.

Comment (matsduf) [reply docs/public/RELEASE.md:6]: But it is still there. Nothing is breaking.

Comment (mattias-p) [reply docs/public/RELEASE.md:6]: I agree that there is a file there but its contents does not reflect the default file. If you want to extract the default profile you need to use the command.

**Review by marc-vanderwal — DISMISSED** [review]

File: docs/public/RELEASE.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+[README]: https://github.com/zonemaster/zonemaster/blob/master/README.md
+[releases]: https://github.com/zonemaster/zonemaster/releases
+
+[ldns-tag]: https://github.com/zonemaster/zonemaster-ldns/releases/tag/v4.0.2
+[engine-tag]: https://github.com/zonemaster/zonemaster-engine/releases/tag/v6.0.0
+[cli-tag]: https://github.com/zonemaster/zonemaster-cli/releases/tag/v7.0.0
+[backend-tag]: https://github.com/zonemaster/zonemaster-backend/releases/tag/v11.2.0
+[gui-tag]: https://github.com/zonemaster/zonemaster-gui/releases/tag/v4.3.0
+
+[zonemaster-Changes]: https://github.com/dotse/zonemaster/blob/master/Changes
```
Comment (marc-vanderwal) [inline docs/public/RELEASE.md:37]:
```suggestion
[zonemaster-Changes]: https://github.com/zonemaster/zonemaster/blob/master/Changes
```

Comment (matsduf) [reply docs/public/RELEASE.md:37]: How did that old reference to "dotse" survice?

Comment (marc-vanderwal) [reply docs/public/RELEASE.md]: Beats me. But it redirects to `zonemaster/zonemaster` just fine.

Comment (matsduf) [reply docs/public/RELEASE.md]: I guess that is why it has not been detected. The referral will be kept by Github until "dotse" is removed or a repository of the same name is created under "dotse".

**Review by marc-vanderwal — DISMISSED** [review]

File: Changes [hunk 4]
```diff
@@ -1,6 +1,33 @@
...
+
+ [Breaking changes]
+ - Removes deprecated profile property in Zonemaster-Engine.
+ - Removes deprecated command line option in Zonemaster-CLI.
+ - Requires database update in Zonemaster-Backend.
+
+ [Features]
+ - Updates implementation of test case Basic01 in Zonemaster-Engine.
+ - Makes IDN names being displayed as both A-label and U-label in Zonemaster-GUI.
+
```
[the remark this answers is not in the document]
Comment (matsduf) [reply Changes:19]: They were omitted on purpose. We have taken a decision only to include changes related to the users. Changes only to unit tests and CI are not to be included. I consider the test zones to be unit test related only, and not related to the product and its behavior.

File: docs/public/RELEASE.md (hunk 2 above)
Comment (matsduf) [reply docs/public/RELEASE.md:6]: Included.

**Review by marc-vanderwal — APPROVED** [review]

File: Changes
[the remark this answers is not in the document]
Comment (matsduf) [reply Changes:1]: I agree. Also the update of the Basic01 specification. And some document updates. I must have been tired when I did this (the last one). I will fix.

File: docs/public/RELEASE.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+ - Removes deprecated command line option in [Zonemaster-CLI].
+ - Requires database update in [Zonemaster-Backend].
+ - Changes how the default profile is to be extracted in [Zonemaster-Engine].
+
+### [Features]
+ - Updates implementation of test case Basic01 in [Zonemaster-Engine].
+ - Makes IDN names being displayed as both A-label and U-label in Zonemaster-GUI.
+
+### [Zonemaster product]
+This version of Zonemaster also consists of the following components. For each component, see its Changes file or Github release notes. See previous releases for recent changes.
```
[the remark this answers is not in the document]
Comment (matsduf) [reply docs/public/RELEASE.md:17]: The last sentence is repeated under the table. I rather remove it.

**Comment by matsduf** [comment]
This is the first time we have `RELEASE.md`. In that we have references to PRs, but they do not become links. Should it be that way?

### [Features]
 - Updates the specification of test case Basic01. (# 1257)
 - Updates implementation of test case Basic01 in [Zonemaster-Engine].
 - Makes IDN names being displayed as both A-label and U-label in [Zonemaster-GUI].

or

### [Features]
 - Updates the specification of test case Basic01. ([#1257])
 - Updates implementation of test case Basic01 in [Zonemaster-Engine].
 - Makes IDN names being displayed as both A-label and U-label in [Zonemaster-GUI].

[#1257]: https://github.com/zonemaster/zonemaster#1257

File: Changes [hunk 6]
```diff
@@ -1,6 +1,39 @@
...
+v2024.1 2024-07-01
+
+ [Release information]
+ - Translations have not been fully updated in this release. They will
+   be updated in an upcoming release.
+
+ [Breaking changes]
+ - Removes deprecated profile property in Zonemaster-Engine.
+ - Removes deprecated command line option in Zonemaster-CLI.
+ - Requires database update in Zonemaster-Backend.
```
Comment (mattias-p) [inline Changes:13]: The database migration should be moved to the Release information heading. In the Backend release notes it's listed under Release information, and I think that's the correct place. The database is not a public interface.

**Comment by matsduf** [comment]
Changes of unit tests are not mentioned in the Changes file. Aren't the test files of the same type?

File: Changes (hunk 6 above)
Comment (matsduf) [reply Changes]: OK, fixat.

File: Changes (hunk 4 above)
[the remark this answers is not in the document]
Comment (matsduf) [reply Changes:27]: I have updated those. Some years ago I added all PRs to the Changes file, also those only related with unit test or Travis updates. But then someone brought that up that only user relevant changes should be included. It was, in a way, simpler when all PRs were included.

**Review by mattias-p — APPROVED** [review]

**Pull request merged**